components/kyma-environment-broker/cmd/environmentscleanup/app_builder.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dlmiddlecote/sqlstats"
+	"github.com/gocraft/dbr"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/common/gardener"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/environmentscleanup"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/events"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/provisioner"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/schema-migrator/cleaner"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	"github.com/vrischmann/envconfig"
+	"golang.org/x/oauth2/clientcredentials"
+	"k8s.io/client-go/dynamic"
+)
+
+type config struct {
+	MaxAgeHours   time.Duration `envconfig:"default=24h"`
+	LabelSelector string        `envconfig:"default=owner.do-not-delete"`
+	Gardener      gardener.Config
+	Database      storage.Config
+	Broker        broker.ClientConfig
+	Provisioner   provisionerConfig
+}
+
+type AppBuilder struct {
+	cfg         config
+	shootClient dynamic.ResourceInterface
+	db          storage.BrokerStorage
+	conn        *dbr.Connection
+	logger      *logrus.Logger
+
+	brokerClient      *broker.Client
+	provisionerClient provisioner.Client
+}
+
+type Job interface {
+	Run() error
+}
+
+func NewAppBuilder() AppBuilder {
+	cfg := config{}
+	err := envconfig.InitWithPrefix(&cfg, "APP")
+	if err != nil {
+		FatalOnError(fmt.Errorf("while loading cleanup config: %w", err))
+	}
+
+	return AppBuilder{}
+}
+
+func (b *AppBuilder) withGardenerClient() {
+	clusterCfg, err := gardener.NewGardenerClusterConfig(b.cfg.Gardener.KubeconfigPath)
+	if err != nil {
+		FatalOnError(fmt.Errorf("while creating Gardener cluster config: %w", err))
+	}
+	cli, err := dynamic.NewForConfig(clusterCfg)
+	if err != nil {
+		FatalOnError(fmt.Errorf("while creating Gardener client: %w", err))
+	}
+	gardenerNamespace := fmt.Sprintf("garden-%s", b.cfg.Gardener.Project)
+	b.shootClient = cli.Resource(gardener.ShootResource).Namespace(gardenerNamespace)
+}
+
+func (b *AppBuilder) withBrokerClient() {
+	ctx := context.Background()
+	b.brokerClient = broker.NewClient(ctx, b.cfg.Broker)
+	b.provisionerClient = provisioner.NewProvisionerClient(b.cfg.Provisioner.URL, b.cfg.Provisioner.QueryDumping)
```

Comment:
If we separate clients in builder calls why not `withProvisionerClient`. Or single method `withBrokerAndProvisionerClients`.